File: RedTeam/Perl/port_scanner.pl
#!/usr/bin/perl

# Author: Forzax HX
# Perl port scanner

use strict;
use warnings;
use Net::IP;
use Socket;

my $ip = shift || '127.0.0.1';
my $port = shift || 21;
my $proto = 'tcp';
my $timeout = 5;
my $cnt = 0;
my $msg = "Puerto $port en la IP $ip";

socket(SERVER, PF_INET, SOCK_STREAM, $proto) or die "socket: $!";
connect(SERVER, $ip) or die "connect: $!";

while (1) {
  select(SERVER);
  $ | = 1;
  if (sysread(SERVER, $buf, 1) != 1) {
    die "error reading from socket: $!";
  }
  if ($buf = ~m / ^ ([0 - 9] + ) $ / ) {
    $cnt++;
    printf("%s", $msg);
    }
    else {
      die "invalid input";
    }
  }

File: RedTeam/Python/port_scanner.py
#!/usr/bin/python

# Author: ForzaxHX
# Python port scanner

import socket as sk

ip = input("Introduzca la IP la cual desea escanear: ")

for port in range(1,65535):
    try:
        s=sk.socket(sk.AF_INET, sk.SOCK_STREAM)
        s.settimeout(100)
        s.connect((ip, port))
        print (str(port)+": OPENED")
        s.close
    except: 
        print (str(port)+": CLOSED")
        continue
